pkg/proxy/proxier.go

```diff
@@ -791,3 +793,7 @@ func (proxier *Proxier) iptablesHostPortalArgs(destIP net.IP, destPort int, prot
 	args = append(args, "-j", "DNAT", "--to-destination", net.JoinHostPort(proxyIP.String(), strconv.Itoa(proxyPort)))
 	return args
 }
+
+func isClosedError(err error) bool {
+	return strings.Contains(err.Error(), "use of closed network connection")
```

Comment:
It may be worth asking the Go team if they are open to extensions here, or
if they think string comparison is the right way to detect this.

On Sat, Apr 11, 2015 at 11:05 PM, Xiang Li <notifications@github.com> wrote:

> In pkg/proxy/proxier.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6737#discussion_r28201039>
> :
>
> > @@ -791,3 +793,7 @@ func (proxier *Proxier) iptablesHostPortalArgs(destIP net.IP, destPort int, prot
> >  	args = append(args, "-j", "DNAT", "--to-destination", net.JoinHostPort(proxyIP.String(), strconv.Itoa(proxyPort)))
> >  	return args
> >  }
> > +
> > +func isClosedError(err error) bool {
> > +	return strings.Contains(err.Error(), "use of closed network connection")
>
> Well... I really want to use the error type instead of the string. But
> this most of the net error is defined privately in go.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6737/files#r28201039>
> .
>